src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(name) {
+    if (name instanceof Herbivore) {
+      const find = Animal.alive.find(animal => animal === name);
+      const findIndex = Animal.alive.findIndex(animal => animal === name);
+
+      if (find.hidden === false) {
+        find.health -= 50;
+      }
+
+      if (name.health <= 0) {
+        Animal.alive.splice(findIndex, 1);
```

Comment:
Checklist says that we can improve our code if we use filter method to delete an animal from the array, we don't need to search for index, and then delete it using splice.